content/en/blog/_posts/2024-06-19-Honor-PV-Reclaim-Policy-Beta.md

```diff
@@ -0,0 +1,194 @@
+---
+layout: blog
+title: 'Kubernetes 1.31: Prevent PersistentVolume leaks when deleting out of order'
+date: 2024-XX-XX
+slug: kubernetes-1-31-prevent-persistentvolume-leaks-when-deleting-out-of-order
+author: >
+  Deepak Kinni (Broadcom)
+---
+
+[PersistentVolume](/docs/concepts/storage/persistent-volumes/) (or PVs for short) are
+associated with [Reclaim Policy](/docs/concepts/storage/persistent-volumes/#reclaim-policy).
+The Reclaim Policy is used to determine the actions that need to be taken by the storage
+backend on deletion of the PVC Bound to a PV.
+When the reclaim policy is `Delete`, the expectation is that the storage backend
+releases the storage resource allocated for the PV. In essence, the reclaim
+policy needs to be honored on PV deletion.
+
+With the recent Kubernetes v1.31 release, a beta feature lets you configure your
+cluster to behave that way and honor the configured reclaim policy.
+
+
+## How did reclaim work in previous Kubernetes releases?
+
+[PersistentVolumeClaim](/docs/concepts/storage/persistent-volumes/#Introduction) (or PVC for short) is
+a user's request for storage. A PV and PVC are considered [Bound](/docs/concepts/storage/persistent-volumes/#Binding)
+if a newly created PV or a matching PV is found. The PVs themselves are
+backed by a volume allocated by the storage backend.
+
+Normally, if the volume is to be deleted, then the expectation is to delete the
+PVC for a bound PV-PVC pair. However, there are no restrictions on deleting a PV
+before deleting a PVC.
+
+First, I'll demonstrate the behavior for clusters running an older version of Kubernetes.
+
+#### Retrieve a PVC that is bound to a PV
+
+Retrieve an existing PVC `example-vanilla-block-pvc`
+```
+kubectl get pvc example-vanilla-block-pvc
+```
+The following output shows the PVC and its `Bound` PV; the PV is shown under the `VOLUME` column:
+```
+NAME                        STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS               AGE
+example-vanilla-block-pvc   Bound    pvc-6791fdd4-5fad-438e-a7fb-16410363e3da   5Gi        RWO            example-vanilla-block-sc   19s
+```
+
+#### Delete PV
+
+When I try to delete a bound PV, the cluster blocks and the `kubectl` tool does
```

Comment:
We may want to say that the kubectl session is blocked rather than the cluster is blocked, right?